src/mpi/comm/comm_split_type.c

```diff
@@ -217,16 +258,45 @@ int MPIR_Comm_split_type_node_topo(MPIR_Comm * user_comm_ptr, int split_type, in
 
     /* initially point to HWLOC_OBJ_TYPE_MAX and then see if there is
      * an info argument pointing to a different object */
-    obj_type = HWLOC_OBJ_TYPE_MAX;
+    info_obj_type.obj_type = HWLOC_OBJ_TYPE_MAX;
+    info_obj_type.device_id[0]='\0';
     if (info_ptr) {
         MPIR_Info_get_impl(info_ptr, SHMEM_INFO_KEY, MPI_MAX_INFO_VAL, hintval, &flag);
         if (flag) {
             /* try to find the info value in the processor object
              * table */
             for (i = 0; shmem_processor_info[i].val; i++) {
                 if (!strcmp(shmem_processor_info[i].val, hintval)) {
-                    obj_type = shmem_processor_info[i].obj_type;
-                    break;
+                		info_obj_type.obj_type = shmem_processor_info[i].obj_type;
+                		break;
+                }
+            }
+
+            /* if we have not found it yet, try to see if the value
+             * matches any of our known strings.  right now, we only
+             * understand "cache". */
+            if (info_obj_type.obj_type == HWLOC_OBJ_TYPE_MAX) {
+            		const char *d = NULL;
+            		if (!strncmp(hintval, "pci:", strlen("pci:"))) {
+            			info_obj_type.obj_type = HWLOC_OBJ_PCI_DEVICE;
+                    d = hintval + strlen("pci:");
+                }
+//                if (!strncmp(hintval, "ib:", strlen("ib:"))) {
+//                    d = hintval + strlen("ib:");
+//                }
+//                if (!strncmp(hintval, "gpu:", strlen("gpu:"))) {
+//                    d = hintval + strlen("ib:");
+//                }
```

Comment:
This is incomplete.  If this patch is not dealing with `ib` and `gpu`, you should delete that code instead of commenting it out.  And, for comments, please use C89-style comments to follow the rest of the MPICH coding style, i.e., `/* comment */`.